### PR TITLE
refactor: creditclock as facet

### LINKED ISSUE #772: Move `CreditClock` to diamond facets
We've migrated most of our contracts to the diamond proxy implementation. But not all of them.

What should be done:
- refactor [CreditClock](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/core/CreditClock.sol) to be part of the diamond [facets](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/src/dollar/facets) 
- add `CreditClock` to the [deployment script](https://github.com/ubiquity/ubiquity-dollar/tree/development/packages/contracts/scripts/deploy/dollar/solidityScripting)
- refactor `CreditClock` [tests](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/test/dollar/core/CreditClock.t.sol) accordingly

### PR DESCRIPTION
**Pull request by molecula451**
Resolves #772 

```
Running 5 tests for test/diamond/facets/CreditClockFacet.t.sol:CreditClockFacetTest
[PASS] test() (gas: 194908653)
[PASS] testGetRate_ShouldRevert_WhenBlockIsInThePast() (gas: 9111)
[PASS] testSetManager_ShouldRevert_WhenNotAdmin() (gas: 21674)
[PASS] testSetManager_ShouldSetDiamond() (gas: 27406)
[PASS] testSetRatePerBlock_ShouldRevert_WhenNotAdmin() (gas: 22105)
Test result: ok. 5 passed; 0 failed; finished in 1.29s
```

## run-latest.json
```
{
      "hash": "0xc8b0c9a50a5071301e158dd2dd9b9f662834a1168347411b33f792b29c8cb1ea",
      "transactionType": "CREATE",
      "contractName": "CreditClockFacet",
      "contractAddress": "0xfA68c64e5782386F4c59b189CDca4771d083a53C",
      "function": null,
      "arguments": [
        "0x0000000000000000000000000000000000000000",
        "0x00000000000000000000000000000000",
        "0x00000000000000000000000000000000"
      ],
      "transaction": {
        "type": "0x02",
        "from": "0xa18e35a6e821aadc80afd132ffa72879f999f2fc",
        "gas": "0x363aed",
        "value": "0x0",
        "data": "0x60a0604052620000106001620000e2565b6001600160801b0319166080523480156200002a57600080fd5b506040516200332c3803806200332c8339810160408190526200004d916200021a565b600080546001600160a01b0319166001600160a01b038516179055436001819055608083811c600160801b84831c8102919091176002819055604080519485526001600160801b031982851b811660208701529290910490921b16908201527f2a888d68449437a913e711ed7752edf8edbeff71a954fc42d427c87cca1817199060600160405180910390a150505062000271565b600081600003620000f557506000919050565b816000620001038262000154565b905060708110156200011e578060700382901b915062000132565b607081111562000132576070810382901c91505b613fff0160701b6001600160701b03919091161760801b92915050565b919050565b60008082116200016357600080fd5b6000600160801b83106200017957608092831c92015b6801000000000000000083106200019257604092831c92015b6401000000008310620001a757602092831c92015b620100008310620001ba57601092831c92015b6101008310620001cc57600892831c92015b60108310620001dd57600492831c92015b60048310620001ee57600292831c92015b60028310620001fb576001015b92915050565b80516001600160801b0319811681146200014f57600080fd5b6000806000606084860312156200023057600080fd5b83516001600160a01b03811681146200024857600080fd5b9250620002586020850162000201565b9150620002686040850162000201565b90509250925092565b608051613098620002946000396000818161028b015261037701526130986000f3fe608060405234801561001057600080fd5b50600436106100885760003560e01c806397d937871161005b57806397d9378714610153578063bf55cd6914610160578063d0ebdbe714610175578063d50095841461018857600080fd5b806313007d551461008d57806334a4b0a5146100d75780635776409414610129578063726839f11461013c575b600080fd5b6000546100ad9073ffffffffffffffffffffffffffffffffffffffff1681565b60405173ffffffffffffffffffffffffffffffffffffffff90911681526020015b60405180910390f35b6002546100f890700100000000000000000000000000000000900460801b81565b6040517fffffffffffffffffffffffffffffffff0000000000000000000000000000000090911681526020016100ce565b6100f8610137366004612f10565b6101a6565b61014560015481565b6040519081526020016100ce565b6002546100f89060801b81565b61017361016e366004612f29565b6103d1565b005b610173610183366004612f72565b610594565b60005473ffffffffffffffffffffffffffffffffffffffff166100ad565b6000816000036101b85743915061024d565b4382101561024d576040517f08c379a000000000000000000000000000000000000000000000000000000000815260206004820152603260248201527f437265646974436c6f636b3a20626c6f636b206e756d626572206d757374206e60448201527f6f7420626520696e2074686520706173742e000000000000000000000000000060648201526084015b60405180910390fd5b6002546103cb9061039d906103549061032d906102ef906102c8907fffffffffffffffffffffffffffffffff000000000000000000000000000000007f00000000000000000000000000000000000000000000000000000000000000001690700100000000000000000000000000000000900460801b6106f9565b7fffffffffffffffffffffffffffffffff0000000000000000000000000000000016610ce4565b610305600154886103009190612fa8565b610f99565b7fffffffffffffffffffffffffffffffff000000000000000000000000000000001690611007565b7fffffffffffffffffffffffffffffffff000000000000000000000000000000001661142e565b7fffffffffffffffffffffffffffffffff000000000000000000000000000000007f00000000000000000000000000000000000000000000000000000000000000001690612a39565b60025460801b7fffffffffffffffffffffffffffffffff000000000000000000000000000000001690611007565b92915050565b6000546040517f91d148540000000000000000000000000000000000000000000000000000000081527faf1a415cb2281de448f1771a3c8144f554e6e38bb3bc1acc8218c01a5d75721d600482015233602482015273ffffffffffffffffffffffffffffffffffffffff909116906391d1485490604401602060405180830381865afa158015610465573d6000803e3d6000fd5b505050506040513d601f19601f820116820180604052508101906104899190612fe2565b6104ef576040517f08c379a000000000000000000000000000000000000000000000000000000000815260206004820152601660248201527f437265646974436c6f636b3a206e6f742061646d696e000000000000000000006044820152606401610244565b6104f8436101a6565b436001819055608091821c70010000000000000000000000000000000084841c8102919091176002819055604080519384527fffffffffffffffffffffffffffffffff0000000000000000000000000000000082861b811660208601529290910490931b16918101919091527f2a888d68449437a913e711ed7752edf8edbeff71a954fc42d427c87cca1817199060600160405180910390a150565b6000546040517f91d148540000000000000000000000000000000000000000000000000000000081527faf1a415cb2281de448f1771a3c8144f554e6e38bb3bc1acc8218c01a5d75721d600482015233602482015273ffffffffffffffffffffffffffffffffffffffff909116906391d1485490604401602060405180830381865afa158015610628573d6000803e3d6000fd5b505050506040513d601f19601f8201168201806040525081019061064c9190612fe2565b6106b2576040517f08c379a000000000000000000000000000000000000000000000000000000000815260206004820152601660248201527f437265646974436c6f636b3a206e6f742061646d696e000000000000000000006044820152606401610244565b600080547fffffffffffffffffffffffff00000000000000000000000000000000000000001673ffffffffffffffffffffffffffffffffffffffff92909216919091179055565b6000617fff60f084811c8216919084901c8116908290036107885780617fff0361077e577fffffffffffffffffffffffffffffffff00000000000000000000000000000000808516908616036107535784925050506103cb565b507f7fff80000000000000000000000000000000000000000000000000000000000091506103cb9050565b84925050506103cb565b80617fff0361079b5783925050506103cb565b6f80000000000000000000000000000000608086901c90811015906dffffffffffffffffffffffffffff1660008490036107d857600193506107ea565b6e010000000000000000000000000000175b6f80000000000000000000000000000000608087901c90811015906dffffffffffffffffffffffffffff1660008590036108275760019450610839565b6e010000000000000000000000000000175b826000036108a0577fffffffffffffffffffffffffffffffff0000000000000000000000000000000088167f8000000000000000000000000000000000000000000000000000000000000000146108905787610893565b60005b96505050505050506103cb565b806000036108f7577fffffffffffffffffffffffffffffffff0000000000000000000000000000000089167f8000000000000000000000000000000000000000000000000000000000000000146108905788610893565b84860382151585151503610a8357607081131561091d57899750505050505050506103cb565b600081131561092f5790811c9061097c565b7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff9081121561096657889750505050505050506103cb565b600081121561097c578060000384901c93508596505b928101926e02000000000000000000000000000084106109a2576001968701969390931c925b86617fff03610a0657846109d6577f7fff0000000000000000000000000000000000000000000000000000000000006109f8565b7fffff0000000000000000000000000000000000000000000000000000000000005b9750505050505050506103cb565b6e010000000000000000000000000000841015610a265760009650610a3a565b6dffffffffffffffffffffffffffff841693505b83607088901b86610a4c576000610a5e565b6f800000000000000000000000000000005b6fffffffffffffffffffffffffffffffff16171760801b9750505050505050506103cb565b6000811315610a9e57600184901b9350600187039650610ab5565b6000811215610ab557600182901b91506001860396505b6070811315610ac75760019150610b50565b6001811315610ae4576001810360018303901c6001019150610b50565b7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff90811215610b155760019350610b50565b7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff811215610b50576001816000030360018503901c60010193505b818410610b61578184039350610b6a565b83820393508294505b83600003610b835750600096506103cb95505050505050565b6000610b8e85612e60565b905080607103610bba57600185901c6dffffffffffffffffffffffffffff169450600188019750610c35565b6070811015610c2157607081900380891115610bf0578086901b6dffffffffffffffffffffffffffff1695508089039850610c1b565b6000987fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff019590951b945b50610c35565b6dffffffffffffffffffffffffffff851694505b87617fff03610c9a5785610c69577f7fff000000000000000000000000000000000000000000000000000000000000610c8b565b7fffff0000000000000000000000000000000000000000000000000000000000005b985050505050505050506103cb565b84607089901b87610cac576000610cbe565b6f800000000000000000000000000000005b6fffffffffffffffffffffffffffffffff16171760801b985050505050505050506103cb565b60006f80000000000000000000000000000000608083901c1115610d2957507f7fff800000000000000000000000000000000000000000000000000000000000919050565b7fffffffffffffffffffffffffffffffff0000000000000000000000000000000082167f3fff00000000000000000000000000000000000000000000000000000000000003610d7a57506000919050565b617fff60f083901c811690819003610d93575090919050565b6dffffffffffffffffffffffffffff608084901c166000829003610dba5760019150610dcc565b6e010000000000000000000000000000175b80600003610dfd57507fffff0000000000000000000000000000000000000000000000000000000000009392505050565b600061406f81613fff8510610e405750600f9290921b91600091507fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc0018401610e8a565b600192506e0100000000000000000000000000008410610e6d5784613ffe039050600f84901b9350610e8a565b6000610e7885612e60565b607f8190039590951b9461406d039150505b836f8000000000000000000000000000000003610ecc578215610eab576001015b6000610eb682612e60565b60700390508082901b9150808303925050610f43565b600083610eda576000610edd565b60015b60ff1690505b6e010000000000000000000000000000821015610f415793800260ff81901c607f81019190911c947fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff939093019260019290921b9082180190610ee3565b505b806dffffffffffffffffffffffffffff16607083901b84610f65576000610f77565b6f800000000000000000000000000000005b6fffffffffffffffffffffffffffffffff16171760801b979650505050505050565b600081600003610fab57506000919050565b816000610fb782612e60565b90506070811015610fd0578060700382901b9150610fe3565b6070811115610fe3576070810382901c91505b613fff0160701b6dffffffffffffffffffffffffffff919091161760801b92915050565b6000617fff60f084811c8216919084901c81169082900361115c5780617fff036110da577fffffffffffffffffffffffffffffffff0000000000000000000000000000000080851690861603611084575050507f8000000000000000000000000000000000000000000000000000000000000000811682186103cb565b7fffffffffffffffffffffffffffffffff00000000000000000000000000000000858518167f800000000000000000000000000000000000000000000000000000000000000003610753575050508181176103cb565b7f7fffffffffffffffffffffffffffffff00000000000000000000000000000000841660000361112f57507f7fff80000000000000000000000000000000000000000000000000000000000091506103cb9050565b5050507f8000000000000000000000000000000000000000000000000000000000000000811682186103cb565b80617fff036111e7577f7fffffffffffffffffffffffffffffff0000000000000000000000000000000085166000036111ba57507f7fff80000000000000000000000000000000000000000000000000000000000091506103cb9050565b5050507f8000000000000000000000000000000000000000000000000000000000000000821681186103cb565b6dffffffffffffffffffffffffffff608086901c16600083900361120e5760019250611220565b6e010000000000000000000000000000175b6dffffffffffffffffffffffffffff608086901c1660008390036112475760019250611259565b6e010000000000000000000000000000175b8082029150816000036112c3577f8000000000000000000000000000000000000000000000000000000000000000878718166112965760006112b8565b7f80000000000000000000000000000000000000000000000000000000000000005b9450505050506103cb565b9282019260007c0200000000000000000000000000000000000000000000000000000000831015611328577c01000000000000000000000000000000000000000000000000000000008310156113215761131c83612e60565b61132b565b60e061132b565b60e15b905061407081860110156113465760009450600092506113ea565b6140e081860110156113895761407085101561136b57846140700383901c9250611380565b61407085111561138057614070850383901b92505b600094506113ea565b61c0dd81860111156113a357617fff9450600092506113ea565b60708111156113ba576070810383901c92506113cd565b60708110156113cd578060700383901b92505b6dffffffffffffffffffffffffffff831692506140df8186010394505b82607086901b888a186f8000000000000000000000000000000060801b1660801c6fffffffffffffffffffffffffffffffff16171760801b955050505050506103cb565b60006f80000000000000000000000000000000608083901c90811190617fff60f085901c8116916dffffffffffffffffffffffffffff16908214801561147357508015155b156114a257507f7fff800000000000000000000000000000000000000000000000000000000000949350505050565b61400d8211156114e357826114d7577f7fff0000000000000000000000000000000000000000000000000000000000006114da565b60005b95945050505050565b613f7f82101561151757507f3fff000000000000000000000000000000000000000000000000000000000000949350505050565b81600003611528576001915061153a565b6e010000000000000000000000000000175b613fef82111561156d577fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc01182011b61157f565b613fef82101561157f57613fef8290031c5b82801561159d575071406e0000000000000000000000000000000081115b156115ad57506000949350505050565b821580156115cc5750713fffffffffffffffffffffffffffffffffff81115b156115fb57507f7fff000000000000000000000000000000000000000000000000000000000000949350505050565b6fffffffffffffffffffffffffffffffff81169060801c83801561161e57508115155b1561162a579019906001015b6f800000000000000000000000000000008281161561165a5770016a09e667f3bcc908b2fb1366ea957d3e0260801c5b6f40000000000000000000000000000000831615611689577001306fe0a31b7152de8d5a46305c85edec0260801c5b6f200000000000000000000000000000008316156116b8577001172b83c7d517adcdf7c8c50eb14a791f0260801c5b6f100000000000000000000000000000008316156116e75770010b5586cf9890f6298b92b71842a983630260801c5b6f08000000000000000000000000000000831615611716577001059b0d31585743ae7c548eb68ca417fd0260801c5b6f0400000000000000000000000000000083161561174557700102c9a3e778060ee6f7caca4f7a29bde80260801c5b6f020000000000000000000000000000008316156117745770010163da9fb33356d84a66ae336dcdfa3f0260801c5b6f010000000000000000000000000000008316156117a357700100b1afa5abcbed6129ab13ec11dc95430260801c5b6e8000000000000000000000000000008316156117d15770010058c86da1c09ea1ff19d294cf2f679b0260801c5b6e4000000000000000000000000000008316156117ff577001002c605e2e8cec506d21bfc89a23a00f0260801c5b6e20000000000000000000000000000083161561182d57700100162f3904051fa128bca9c55c31e5df0260801c5b6e10000000000000000000000000000083161561185b577001000b175effdc76ba38e31671ca9397250260801c5b6e08000000000000000000000000000083161561188957700100058ba01fb9f96d6cacd4b180917c3d0260801c5b6e0400000000000000000000000000008316156118b75770010002c5cc37da9491d0985c348c68e7b30260801c5b6e0200000000000000000000000000008316156118e5577001000162e525ee054754457d59952920260260801c5b6e0100000000000000000000000000008316156119135770010000b17255775c040618bf4a4ade83fc0260801c5b6d8000000000000000000000000000831615611940577001000058b91b5bc9ae2eed81e9b7d4cfab0260801c5b6d400000000000000000000000000083161561196d57700100002c5c89d5ec6ca4d7c8acc017b7c90260801c5b6d200000000000000000000000000083161561199a5770010000162e43f4f831060e02d839a9d16d0260801c5b6d10000000000000000000000000008316156119c757700100000b1721bcfc99d9f890ea069117630260801c5b6d08000000000000000000000000008316156119f45770010000058b90cf1e6d97f9ca14dbcc16280260801c5b6d0400000000000000000000000000831615611a21577001000002c5c863b73f016468f6bac5ca2b0260801c5b6d0200000000000000000000000000831615611a4e57700100000162e430e5a18f6119e3c02282a50260801c5b6d0100000000000000000000000000831615611a7b577001000000b1721835514b86e6d96efd1bfe0260801c5b6c80000000000000000000000000831615611aa757700100000058b90c0b48c6be5df846c5b2ef0260801c5b6c40000000000000000000000000831615611ad35770010000002c5c8601cc6b9e94213c72737a0260801c5b6c20000000000000000000000000831615611aff577001000000162e42fff037df38aa2b219f060260801c5b6c10000000000000000000000000831615611b2b5770010000000b17217fba9c739aa5819f44f90260801c5b6c08000000000000000000000000831615611b57577001000000058b90bfcdee5acd3c1cedc8230260801c5b6c04000000000000000000000000831615611b8357700100000002c5c85fe31f35a6a30da1be500260801c5b6c02000000000000000000000000831615611baf5770010000000162e42ff0999ce3541b9fffcf0260801c5b6c01000000000000000000000000831615611bdb57700100000000b17217f80f4ef5aadda455540260801c5b6b800000000000000000000000831615611c065770010000000058b90bfbf8479bd5a81b51ad0260801c5b6b400000000000000000000000831615611c31577001000000002c5c85fdf84bd62ae30a74cc0260801c5b6b200000000000000000000000831615611c5c57700100000000162e42fefb2fed257559bdaa0260801c5b6b100000000000000000000000831615611c87577001000000000b17217f7d5a7716bba4a9ae0260801c5b6b080000000000000000000000831615611cb257700100000000058b90bfbe9ddbac5e109cce0260801c5b6b040000000000000000000000831615611cdd5770010000000002c5c85fdf4b15de6f17eb0d0260801c5b6b020000000000000000000000831615611d08577001000000000162e42fefa494f1478fde050260801c5b6b010000000000000000000000831615611d335770010000000000b17217f7d20cf927c8e94c0260801c5b6a8000000000000000000000831615611d5d577001000000000058b90bfbe8f71cb4e4b33d0260801c5b6a4000000000000000000000831615611d8757700100000000002c5c85fdf477b662b269450260801c5b6a2000000000000000000000831615611db15770010000000000162e42fefa3ae53369388c0260801c5b6a1000000000000000000000831615611ddb57700100000000000b17217f7d1d351a389d400260801c5b6a0800000000000000000000831615611e055770010000000000058b90bfbe8e8b2d3d4ede0260801c5b6a0400000000000000000000831615611e2f577001000000000002c5c85fdf4741bea6e77e0260801c5b6a0200000000000000000000831615611e5957700100000000000162e42fefa39fe95583c20260801c5b6a0100000000000000000000831615611e83577001000000000000b17217f7d1cfb72b45e10260801c5b6980000000000000000000831615611eac57700100000000000058b90bfbe8e7cc35c3f00260801c5b6940000000000000000000831615611ed55770010000000000002c5c85fdf473e242ea380260801c5b6920000000000000000000831615611efe577001000000000000162e42fefa39f02b772c0260801c5b6910000000000000000000831615611f275770010000000000000b17217f7d1cf7d83c1a0260801c5b6908000000000000000000831615611f50577001000000000000058b90bfbe8e7bdcbe2e0260801c5b6904000000000000000000831615611f7957700100000000000002c5c85fdf473dea871f0260801c5b6902000000000000000000831615611fa25770010000000000000162e42fefa39ef44d910260801c5b6901000000000000000000831615611fcb57700100000000000000b17217f7d1cf79e9490260801c5b68800000000000000000831615611ff35770010000000000000058b90bfbe8e7bce5440260801c5b6840000000000000000083161561201b577001000000000000002c5c85fdf473de6eca0260801c5b6820000000000000000083161561204357700100000000000000162e42fefa39ef366f0260801c5b6810000000000000000083161561206b577001000000000000000b17217f7d1cf79afa0260801c5b6808000000000000000083161561209357700100000000000000058b90bfbe8e7bcd6d0260801c5b680400000000000000008316156120bb5770010000000000000002c5c85fdf473de6b20260801c5b680200000000000000008316156120e3577001000000000000000162e42fefa39ef3580260801c5b6801000000000000000083161561210b5770010000000000000000b17217f7d1cf79ab0260801c5b678000000000000000831615612132577001000000000000000058b90bfbe8e7bcd50260801c5b67400000000000000083161561215957700100000000000000002c5c85fdf473de6a0260801c5b6720000000000000008316156121805770010000000000000000162e42fefa39ef340260801c5b6710000000000000008316156121a757700100000000000000000b17217f7d1cf7990260801c5b6708000000000000008316156121ce5770010000000000000000058b90bfbe8e7bcc0260801c5b6704000000000000008316156121f5577001000000000000000002c5c85fdf473de50260801c5b67020000000000000083161561221c57700100000000000000000162e42fefa39ef20260801c5b670100000000000000831615612243577001000000000000000000b17217f7d1cf780260801c5b668000000000000083161561226957700100000000000000000058b90bfbe8e7bb0260801c5b664000000000000083161561228f5770010000000000000000002c5c85fdf473dd0260801c5b66200000000000008316156122b5577001000000000000000000162e42fefa39ee0260801c5b66100000000000008316156122db5770010000000000000000000b17217f7d1cf60260801c5b6608000000000000831615612301577001000000000000000000058b90bfbe8e7a0260801c5b660400000000000083161561232757700100000000000000000002c5c85fdf473c0260801c5b660200000000000083161561234d5770010000000000000000000162e42fefa39d0260801c5b660100000000000083161561237357700100000000000000000000b17217f7d1ce0260801c5b658000000000008316156123985770010000000000000000000058b90bfbe8e60260801c5b654000000000008316156123bd577001000000000000000000002c5c85fdf4720260801c5b652000000000008316156123e257700100000000000000000000162e42fefa380260801c5b65100000000000831615612407577001000000000000000000000b17217f7d1b0260801c5b6508000000000083161561242c57700100000000000000000000058b90bfbe8d0260801c5b650400000000008316156124515770010000000000000000000002c5c85fdf460260801c5b65020000000000831615612476577001000000000000000000000162e42fefa20260801c5b6501000000000083161561249b5770010000000000000000000000b17217f7d00260801c5b6480000000008316156124bf577001000000000000000000000058b90bfbe70260801c5b6440000000008316156124e357700100000000000000000000002c5c85fdf30260801c5b6420000000008316156125075770010000000000000000000000162e42fef90260801c5b64100000000083161561252b57700100000000000000000000000b17217f7c0260801c5b64080000000083161561254f5770010000000000000000000000058b90bfbd0260801c5b640400000000831615612573577001000000000000000000000002c5c85fde0260801c5b64020000000083161561259757700100000000000000000000000162e42fee0260801c5b6401000000008316156125bb577001000000000000000000000000b17217f60260801c5b63800000008316156125de57700100000000000000000000000058b90bfa0260801c5b63400000008316156126015770010000000000000000000000002c5c85fc0260801c5b6320000000831615612624577001000000000000000000000000162e42fd0260801c5b63100000008316156126475770010000000000000000000000000b17217e0260801c5b630800000083161561266a577001000000000000000000000000058b90be0260801c5b630400000083161561268d57700100000000000000000000000002c5c85e0260801c5b63020000008316156126b05770010000000000000000000000000162e42e0260801c5b63010000008316156126d357700100000000000000000000000000b172160260801c5b628000008316156126f55770010000000000000000000000000058b90a0260801c5b62400000831615612717577001000000000000000000000000002c5c840260801c5b6220000083161561273957700100000000000000000000000000162e410260801c5b6210000083161561275b577001000000000000000000000000000b17200260801c5b6208000083161561277d57700100000000000000000000000000058b8f0260801c5b6204000083161561279f5770010000000000000000000000000002c5c70260801c5b620200008316156127c1577001000000000000000000000000000162e30260801c5b620100008316156127e35770010000000000000000000000000000b1710260801c5b618000831615612804577001000000000000000000000000000058b80260801c5b61400083161561282557700100000000000000000000000000002c5b0260801c5b6120008316156128465770010000000000000000000000000000162d0260801c5b61100083161561286757700100000000000000000000000000000b160260801c5b6108008316156128885770010000000000000000000000000000058a0260801c5b6104008316156128a9577001000000000000000000000000000002c40260801c5b6102008316156128ca577001000000000000000000000000000001610260801c5b6101008316156128eb577001000000000000000000000000000000b00260801c5b608083161561290b577001000000000000000000000000000000570260801c5b604083161561292b5770010000000000000000000000000000002b0260801c5b602083161561294b577001000000000000000000000000000000150260801c5b601083161561296b5770010000000000000000000000000000000a0260801c5b600883161561298b577001000000000000000000000000000000040260801c5b60048316156129ab577001000000000000000000000000000000010260801c5b846129d357600f81901c6dffffffffffffffffffffffffffff169050613fff82019150612a26565b613ffe82116129ff57600f81901c6dffffffffffffffffffffffffffff16905081613fff039150612a26565b6000917fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc011011c5b60709190911b1760801b95945050505050565b6000617fff60f084811c8216919084901c811690829003612a885780617fff0361112f57507f7fff80000000000000000000000000000000000000000000000000000000000091506103cb9050565b80617fff03612b0f577dffffffffffffffffffffffffffff00000000000000000000000000000000841615612ae257507f7fff80000000000000000000000000000000000000000000000000000000000091506103cb9050565b5050508082187f8000000000000000000000000000000000000000000000000000000000000000166103cb565b7f7fffffffffffffffffffffffffffffff000000000000000000000000000000008416600003612bdd577f7fffffffffffffffffffffffffffffff000000000000000000000000000000008516600003612b8e57507f7fff80000000000000000000000000000000000000000000000000000000000091506103cb9050565b5050508082187f8000000000000000000000000000000000000000000000000000000000000000167f7fff000000000000000000000000000000000000000000000000000000000000176103cb565b6dffffffffffffffffffffffffffff608085901c166000829003612c045760019150612c16565b6e010000000000000000000000000000175b6dffffffffffffffffffffffffffff608087901c166000849003612c7e578015612c79576000612c4582612e60565b6001955060e2039384017fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff8e01939190911b90505b612c93565b6e0100000000000000000000000000001760721b5b818181612ca257612ca2613004565b04905080600003612cdd577f8000000000000000000000000000000000000000000000000000000000000000878718166112965760006112b8565b6d1000000000000000000000000000811015612cfb57612cfb613033565b60006e080000000000000000000000000000821015612d5b576e040000000000000000000000000000821015612d50576e020000000000000000000000000000821015612d49576070612d53565b6071612d53565b60725b60ff16612d64565b612d6482612e60565b905083614071018186011115612d8257617fff945060009150612e1c565b83818601613ffc011015612d9d576000945060009150612e1c565b83818601613f8c011015612dea578385613ffc011115612dc8578385613ffc010382901b9150612de1565b8385613ffc011015612de157613ffc8585030382901c91505b60009450612e1c565b6070811115612dfd576070810382901c91505b6dffffffffffffffffffffffffffff8216915083818601613f8d010394505b81607086901b888a186f8000000000000000000000000000000060801b1660801c6fffffffffffffffffffffffffffffffff16171760801b955050505050506103cb565b6000808211612e6e57600080fd5b60007001000000000000000000000000000000008310612e9057608092831c92015b680100000000000000008310612ea857604092831c92015b6401000000008310612ebc57602092831c92015b620100008310612ece57601092831c92015b6101008310612edf57600892831c92015b60108310612eef57600492831c92015b60048310612eff57600292831c92015b600283106103cb5760010192915050565b600060208284031215612f2257600080fd5b5035919050565b600060208284031215612f3b57600080fd5b81357fffffffffffffffffffffffffffffffff0000000000000000000000000000000081168114612f6b57600080fd5b9392505050565b600060208284031215612f8457600080fd5b813573ffffffffffffffffffffffffffffffffffffffff81168114612f6b57600080fd5b818103818111156103cb577f4e487b7100000000000000000000000000000000000000000000000000000000600052601160045260246000fd5b600060208284031215612ff457600080fd5b81518015158114612f6b57600080fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052601260045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052600160045260246000fdfea264697066735822122009d009d281bbc95ad134caf4234939cbcb39efedbb6e5f7209b45a6dacb3ca2d64736f6c63430008130033000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
        "nonce": "0x1d1",
        "accessList": []
      },
      "additionalContracts": [],
      "isFixedGasLimit": false
    }
```

### CODE REVIEW

File: packages/contracts/scripts/deploy/dollar/solidityScripting/00_Constants.s.sol [hunk 1]
```diff
@@ -130,6 +130,6 @@ contract Constants is Script {
     ];
 
     //CreditClockFacet
-    bytes16 _rateStartValue = (0x00);
-    bytes16 _ratePerBlock = (0x00);
+    bytes16 constant RATE_START_VALUE = (0x00);
```
Comment (molecula451) [inline packages/contracts/scripts/deploy/dollar/solidityScripting/00_Constants.s.sol:133]: to be update for production values

**Review by rndquu — CHANGES_REQUESTED** [review]
Facets can't have state variables (at least in our project).

Check this [facet](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/CollectableDustFacet.sol). All of the state variables for the `CollectableDustFacet` contract are managed in a separate library [here](https://github.com/ubiquity/ubiquity-dollar/blob/36b920515eebcf4d8db08fc51d36f08e571755b4/packages/contracts/src/dollar/libraries/LibCollectableDust.sol#L29) starting from special [slot](https://github.com/ubiquity/ubiquity-dollar/blob/36b920515eebcf4d8db08fc51d36f08e571755b4/packages/contracts/src/dollar/libraries/LibCollectableDust.sol#L45).

With the current `CreditClockFacet` [implementation](https://github.com/ubiquity/ubiquity-dollar/blob/2519eb41248e485e1c0ccdf054f24b8c1a68982d/packages/contracts/src/dollar/facets/CreditClockFacet.sol), in case we add a new facet with state variables then those state variables (from `CreditClockFacet` and a newly added facet) will start to rewrite each other because all of them would start from slot 0 (and keeping in mind that the `Diamond` contract uses `delegatecall()` to call facets).

Pls refactor `CreditClockFacet` to use a library which accesses all of `CreditClockFacet` state variables starting from a special slot `ubiquity.contracts.credit.clock.storage`.

**Comment by rndquu** [comment]
@molecula451 Is it ready for review?

**Comment by molecula451** [comment]
> @molecula451 Is it ready for review?

I'm finishing things up on `CreditClockFacet.t.sol`

**Comment by rndquu** [comment]
@molecula451 Any updates? The bot should've [closed](https://github.com/ubiquity/ubiquity-dollar/issues/772#issuecomment-1704749887) this PR 2 weeks ago

**Comment by molecula451** [comment]
> @molecula451 Any updates? The bot should've [closed](https://github.com/ubiquity/ubiquity-dollar/issues/772#issuecomment-1704749887) this PR 2 weeks ago

I got this stale, i need to tackle a bug on the access control modifier after the implementation, this could be closed tho, if someone else comes with a PR for the issue, it's also good

**Comment by molecula451** [comment]
closing, i will re-open if re-start working on it, & if there is no new PR's the review was helpful tho

**Pull request closed without merging**